inc/packages/admin/namespace.php

```diff
@@ -271,3 +298,59 @@ function maybe_hijack_plugin_info() {
 	Info\render_page( $metadata, $tab, $section );
 	exit;
 }
+
+/**
+ * Maybe hijack the legacy plugin info too.
+ */
+function maybe_hijack_legacy_plugin_info() {
+	$api = plugins_api(
+		'plugin_information',
+		array(
+			'slug' => wp_unslash( $_REQUEST['plugin'] ),
+		)
+	);
+
+	if ( is_wp_error( $api ) ) {
+		wp_die( $api );
+	}
+
+	// Is this a FAIR plugin, actually?
+	if ( ! is_fair_plugin( $api ) ) {
+		return;
+	}
+
+	// Neat! Upgrade it.
+	// We need to convert deeply to objects, so re-encode as JSON.
+	$reencoded = json_decode( json_encode( $api->_fair ), false );
+	$metadata = MetadataDocument::from_data( $reencoded );
+	if ( is_wp_error( $metadata ) ) {
+		wp_die( esc_html( $metadata->get_error_message() ) );
+	}
+
+	$tab = esc_attr( $GLOBALS['tab'] ?? 'plugin-information' );
+	$section = isset( $_REQUEST['section'] ) ? sanitize_key( wp_unslash( $_REQUEST['section'] ) ) : 'description';
+	Info\render_page( $metadata, $tab, $section );
+	exit;
+}
+
+/**
+ * Filters the plugin card description on the Add Plugins screen.
+ *
+ * @param string $description Plugin card description.
+ * @param array $plugin (Legacy) plugin data from the dotorg API.
+ * @return string Plugin card description.
+ */
+function maybe_add_data_to_description( $description, $plugin ) {
+	if ( ! is_fair_plugin( $plugin ) ) {
+		return $description;
+	}
+
+	$did = $plugin['_fair']['id'];
+	$repo_host = Info\get_repository_hostname( $did );
+	if ( empty( $repo_host ) ) {
+		return $description;
+	}
+
+	$description .= '</p><p class="authors"><em>' . sprintf( __( 'Hosted on %s', 'fair' ), esc_html( $repo_host ) ) . '</em>';
```

Comment:
Can we drop a translators comment here so it's clear `%s` isn't, for example, the Repository Host's name?

Just thinking about `Hosted on fair.git-updater.com` vs `Hosted on Git Updater`, and how that could affect a translation, such as the appropriate choice between on/at/from/by/with/etc. for a given locale.